astropy/io/fits/tests/test_hdulist.py

```diff
@@ -868,3 +869,25 @@ def test_proper_error_raised_on_non_fits_file_with_unicode(self):
         # This should raise an IOError because there is no end card.
         with pytest.raises(IOError):
             fits.open(filename)
+
+    # Avoid a spurious pyflakes warning about an unused import
+    # of six by adding a reason string
+    @pytest.mark.skipif(six.PY2, reason='six.PY2')
+    def test_no_resource_warning_raised_on_non_fits_file(self):
+        """
+        Regression test for https://github.com/astropy/astropy/issues/6168
+
+        The ResourceWarning shows up when (in python 3+) you try to
+        open a non-FITS file.
+        """
+        filename = self.temp('not-fits.fits')
+        with open(filename, mode='w') as f:
+            f.write('# header line\n')
+            f.write('0.1 0.2\n')
+
+        # This should raise an IOError because there is no end card.
+        with catch_warnings() as ws:
```

Comment:
It would be good to check that the test fails without your fix: `ResourceWarning` is ignored by default (https://docs.python.org/3/library/warnings.html#default-warning-filters), not sure what's the status with pytest's warning plugin.